src/Controls/src/SourceGen/CodeBehindGenerator.cs

```diff
@@ -293,7 +300,7 @@ static bool GetXamlCompilationProcessingInstruction(XmlDocument xmlDoc)
 			return false;
 		}
 
-		static IEnumerable<(string name, string type, string accessModifier)> GetNamedFields(XmlNode root, XmlNamespaceManager nsmgr, Compilation compilation, IList<XmlnsDefinitionAttribute> xmlnsDefinitionCache)
+		static IEnumerable<(string name, string type, string accessModifier)> GetNamedFields(XmlNode root, XmlNamespaceManager nsmgr, Compilation compilation, IList<XmlnsDefinitionAttribute> xmlnsDefinitionCache, CancellationToken cancellationToken)
```

Comment:
Would it be possible to also get the cancellation token into GetTypeName, since I think that was the hotspot here?

Reply:
There is no loop inside of GetTypeName so the check in the GetNamedFields loop will take care of it.

Reply:
The loop was here if I recall:

https://github.com/dotnet/maui/blob/9459133a620f10416ec8775f2be0afda58cab998/src/Controls/src/SourceGen/CodeBehindGenerator.cs#L365

Since that calls into:

https://github.com/dotnet/maui/blob/9459133a620f10416ec8775f2be0afda58cab998/src/Controls/src/Xaml/XmlTypeXamlExtensions.cs#L80-L89

Which had that nested foreach loop.

Reply:
That nested foreach isn't actually doing that much, it just creates a list of potential full type names, which rather small. The rest gets called a lot more often. This was just about cancelling at strategic places. I'm working on another change which should address the real perf issues in this generator.